src/msg/async/dpdk/dpdk_rte.cc

```diff
@@ -45,16 +45,47 @@ namespace dpdk {
     return std::bitset<CHAR_BIT * sizeof(n)>{n}.count();
   }
 
+  static int xdigit2val(unsigned char c)
+  {
+    int val;
+
+    if (isdigit(c))
+      val = c - '0';
+    else if (isupper(c))
+      val = c - 'A' + 10;
+    else
+      val = c - 'a' + 10;
+    return val;
+  }
+
+  static int coremask_bitcount(const char *coremask)
+  {
+    int count = 0;
+
+    if (coremask[0] == '0' && ((coremask[1] == 'x')
+        || (coremask[1] == 'X')))
+      coremask += 2;
+
+    for (int i = 0; coremask[i] != '\0'; i++) {
+      char c = coremask[i];
+      if (isxdigit(c) == 0)
+        return 0;
+      int val = xdigit2val(c);
+      if (val)
```

Comment:
nit, no need to make it a special case.

Reply:
If val is 0, it does not contain 1, so we will not count it.

Reply:
i think i understand what `bitcount()` is for. but my point is "no need to make it a special case", i.e. it does not make difference if we remove this `if`.

Reply:
get it, thanks